### PR TITLE
Remove redundancy and clean code

### PR DESCRIPTION
**Pull request by faysal-alam-sazim**
### Description of Change
- Used Context provider instead of prop drilling for To Do App page
- Moved all CRUD functions to the context api
- Also moved history state and undo redo function to the context
### Associated Ticket Link:
- https://trello.com/c/ZvyL70fD

### CODE REVIEW

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Please wait for Ragib bhais approval before merging

**Review by ragib-hossain-sazim — APPROVED** [review]
Approved but there seems to be some unnecessary commits. Please make sure the PR is rebased properly before merging.

**Pull request merged**
